site/smart-accounts/signers/guides/dfns.md

```diff
@@ -0,0 +1,129 @@
+---
+outline: deep
+head:
+  - - meta
+    - property: og:title
+      content: Dfns
+  - - meta
+    - name: description
+      content: Guide for using Dfns as a signer
+  - - meta
+    - property: og:description
+      content: Guide for using Dfns as a signer
+---
+
+# Dfns Integration Guide
+
+[Dfns](https://www.dfns.co) is an MPC/TSS Wallet-as-a-Service API/SDK provider. We have optimized the balance of security and UX by deploying key shares into a decentralized network on the backend while enabling wallet access via biometric open standards on the frontend like Webauthn. [Ping us](https://www.dfns.co/learn-more) to set up a sandbox environment to get started.
+
+Dfns seamlessly integrates with Account Abstraction by signing User Operations. We've created a full example of a gasless transaction via a paymaster [in our SDK](https://github.com/dfns/dfns-sdk-ts/tree/m/examples/viem/alchemy-aa-gasless) adapted from Alchemy's [sponsor gas example](https://accountkit.alchemy.com/guides/sponsoring-gas.html). Clone the repo to get started and see the comments in the [Readme](https://github.com/dfns/dfns-sdk-ts/tree/m/examples/viem/alchemy-aa-gasless/README.md).
+
+## Install Dfns SDK
+
+::: code-group
+
+```bash [npm]
+npm i @dfns/lib-viem @dfns/sdk @dfns/sdk-keysigner
+```
+
+```bash [yarn]
+yarn add @dfns/lib-viem @dfns/sdk @dfns/sdk-keysigner
+```
+
+:::
+
+### Create a Dfns SmartAccountSigner
+
+Setup the Dfns Web3 Provider and wrap it in an AlchemyProvider. See the Dfns example [Readme](https://github.com/dfnsext/typescript-sdk/blob/m/examples/viem/alchemy-aa-gasless/README.md) for detail on populating the environment variables.
+
+```ts
+import {
+  LightSmartContractAccount,
+  getDefaultLightAccountFactoryAddress,
+} from "@alchemy/aa-accounts";
+import { AlchemyProvider } from "@alchemy/aa-alchemy";
+import { LocalAccountSigner } from "@alchemy/aa-core";
+import { DfnsWallet } from "@dfns/lib-viem";
+import { DfnsApiClient } from "@dfns/sdk";
+import { AsymmetricKeySigner } from "@dfns/sdk-keysigner";
+import dotenv from "dotenv";
+import { encodeFunctionData, parseAbi } from "viem";
+import { toAccount } from "viem/accounts";
+import { sepolia } from "viem/chains";
+
+dotenv.config();
+
+const initDfnsWallet = (walletId: string) => {
```

Comment:
can you make this a snippet and add it to `site/snippets/dfns.ts`? feel free to look at the other signers to see how they do this.